applications/FluidDynamicsApplication/custom_elements/d_vms_dem_coupled.cpp

```diff
@@ -0,0 +1,497 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Joaquin Gonzalez-Usua
+//
+
+// Project includes
+#include "includes/cfd_variables.h"
+#include "includes/dem_variables.h"
+#include "includes/checks.h"
+
+// Aplication includes
+#include "d_vms_dem_coupled.h"
+#include "custom_utilities/qsvms_dem_coupled_data.h"
+#include "custom_utilities/fluid_element_utilities.h"
+#include "fluid_dynamics_application_variables.h"
+
+namespace Kratos
+{
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Life cycle
+
+template< class TElementData >
+DVMSDEMCoupled<TElementData>::DVMSDEMCoupled(IndexType NewId):
+    DVMS<TElementData>(NewId)
+{}
+
+template< class TElementData >
+DVMSDEMCoupled<TElementData>::DVMSDEMCoupled(IndexType NewId, const NodesArrayType& ThisNodes):
+    DVMS<TElementData>(NewId,ThisNodes)
+{}
+
+
+template< class TElementData >
+DVMSDEMCoupled<TElementData>::DVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry):
+    DVMS<TElementData>(NewId,pGeometry)
+{}
+
+
+template< class TElementData >
+DVMSDEMCoupled<TElementData>::DVMSDEMCoupled(IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties):
+    DVMS<TElementData>(NewId,pGeometry,pProperties)
+{}
+
+
+template< class TElementData >
+DVMSDEMCoupled<TElementData>::~DVMSDEMCoupled()
+{}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Public Operations
+
+template< class TElementData >
+Element::Pointer DVMSDEMCoupled<TElementData>::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<DVMSDEMCoupled>(NewId, this->GetGeometry().Create(ThisNodes), pProperties);
+}
+
+
+template< class TElementData >
+Element::Pointer DVMSDEMCoupled<TElementData>::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<DVMSDEMCoupled>(NewId, pGeom, pProperties);
+}
+
+template <class TElementData>
+void DVMSDEMCoupled<TElementData>::Calculate(
+    const Variable<double>& rVariable,
+    double& rOutput,
+    const ProcessInfo& rCurrentProcessInfo) {}
+
+template <class TElementData>
+void DVMSDEMCoupled<TElementData>::Calculate(
+    const Variable<array_1d<double, 3>>& rVariable,
+    array_1d<double, 3>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo) {
+    // Lumped projection terms
+    if (rVariable == ADVPROJ) {
+        this->CalculateProjections(rCurrentProcessInfo);
+    }
```

Comment:
An `else` statement calling the base class `Calculate` should be added in case the provided variable it's not implemented in here.